app/CreateDeal.tsx

```diff
@@ -84,13 +147,38 @@ export function CreateDeal(props: {
             if (pricePerUnit === "0") {
               return setErrorMessage("Price cannot be $0");
             }
-
-            console.log(
-              "item:" + itemName,
-              typeof itemName,
-              "price per unit:" + pricePerUnit,
-              typeof pricePerUnit
+            if (selectedUnit === undefined) {
+              return setErrorMessage("Must select a unit");
+            }
+            if (selectedStoreId === undefined) {
+              return setErrorMessage("Must select a store");
+            }
+            if (selectedDate === undefined) {
+              return setErrorMessage("Must select a date");
+            }
+            setErrorMessage("");
+            setAwaitingCreateDealCheck(true);
+            if (
+              await itemAlreadyExists(
+                itemName,
+                Number(pricePerUnit),
+                selectedStoreId
+              )
+            ) {
+              setAwaitingCreateDealCheck(false);
+              return setErrorMessage(
+                "An item with the same details already exists"
+              );
+            }
+            onCreateDealClick();
+            createNewDeal(
+              itemName,
+              Number(pricePerUnit),
+              selectedUnit,
+              new Date(selectedDate),
+              selectedStoreId
             );
+            setAwaitingCreateDealCheck(false);
```

Comment:
`createNewDeal` is an async function but you aren't awaiting it, which means the button will be available to be clicked again before the deal is actually created so someone could press it really fast more than once and create duplicates of the same deal